pkg/apis/application/v1alpha1/application_types.go

```diff
@@ -64,6 +66,37 @@ type ApplicationStatus struct {
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
 }
 
+// Maintainer contains information about an individual or organization that maintains the source code, images, and
+// package for an Application. An Application can have more than one maintainer.
+type Maintainer struct {
+	// Name is the descriptive name of the maintainer.
+	Name string `json:"name,omitempty"`
+
+	// Url could typically be a website address or a mailto: email address.
+	Url string `json:"url,omitempty"`
+}
+
+// Link contains information about an URL to surface documentation, dashboards, etc.
+type Link struct {
+	// Type defines the purpose of the link.
+	Type string `json:"type,omitempty"`
```

Comment:
Should there be a valid set of strings for Type? (e.g. "documentation", "support", "marketing", ...)

Reply:
@deustis On other projects that capture this same kind of information are there cases that capture this form of standardized categorization?

Reply:
This is probably not an enum, although we can recommend some convention around the values of this field, for example `application.k8s.io/support`.

Should we document such convention here, or just keep it this way?

Reply:
I think we should actually remove type. I don't see what value it provides.

Reply:
An example of how these fields are used:
```
links:
- type: support
  description: Community support for Wordpress
  url: https://stackoverflow.com/questions/tagged/wordpress/
- type: info
  description: More information about Wordpress
  url: https://wordpress.org/
- type: license
  url: https://wordpress.org/about/license/
```

The `info` link in this particular Wordpress case could be removed and replaced by the maintainer's URL. We can imagine a case where the same maintainer (i.e. company) has many applications. `info` would be application specific and useful.

Reply:
@mattfarina, no standards that I'm aware of.

We ([Google Cloud Launcher](https://console.cloud.google.com/launcher)) have experience collecting product metadata from software vendors. We don't try to categorize different link types today, but we do see very common recurring patterns. We could try to pull together some data if that would be helpful.

The main reason to consider categorizing links would be able to give them special treatment in UIs. for example, UIs would typically want to feature the "user guide" more prominently than "tutorials", or "community forums".

I do think we should either remove Type, or restrict it to a specific set of values. Otherwise, as a free-form string, it doesn't add much value.

Reply:
I'm removing `Type` as it is the only point of contention. We can continue this discussion in another issue/PR.